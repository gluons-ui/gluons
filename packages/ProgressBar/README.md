## Mandala UI ProgressBar Component
Mandala is a React component library that utilizes a declarative CSS library for styling.

### Installation
Check the [MonoRepo README](https://github.com/mandala-ui/mandala) for installation.

### Description
ProgressBar is a 100% width element that takes a percent prop and renders that much of it in another color.

### Usage
if using the whole library:
```js
import { ProgressBar } from 'mandala';
```
if only using this component:
```js
import ProgressBar from '@mandala-ui/progress-bar';
```
### Props
* `backgroundColor` - the color of the border around the dot
* `color` - the width of the border around the dot
* `large` - the color of the dot
* `percent` - the width of the inner section between the dot and border
* `pill` - the color of the inner section of the outlined dot

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| backgroundColor | string | null    | -          |
| color      | string | false        | -          |
| large      | boolean| 'white'      | -          |
| percent    | number | false        | -          |
| pill       | boolean| false        | -          |


#### TODO:
- [ ] prop to add/remove depth (shadow)
- [ ] borderColor adds a border of relevant color
- [ ] border width

![mandala-bottom](https://user-images.githubusercontent.com/1824267/38281372-32114e5c-375f-11e8-9fbe-e5191b96429c.png)
