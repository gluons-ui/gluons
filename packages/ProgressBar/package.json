{
  "author": "Lucas K Allmon",
  "dependencies": {
    "lodash.clamp": "^4.0.3"
  },
  "description": "Mandala UI Progress Bar Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user",
    "ux"
  ],
  "license": "MIT",
  "main": "lib/ProgressBar.js",
  "module": "es/ProgressBar.js",
  "name": "@mandala-ui/progress-bar",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "1.0.2"
}
