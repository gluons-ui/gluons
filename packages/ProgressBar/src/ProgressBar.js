import React from 'react';
import PropTypes from 'prop-types';
import clamp from 'lodash/clamp';

const ProgressBar = ({
  backgroundColor,
  color,
  large,
  percent,
  pill,
}) => {
  const isLarge = `${large ? 'h2' : 'h1'}`;
  const isPill = `${pill ? 'br-pill' : ''}`;
  const fill = `bg-${color} ${isLarge} ${isPill}`;
  const container = `bg-${backgroundColor} ${isLarge} ${isPill}`;
  return (
    <div className="progress-bar w-100">
      <div className="w-100">
        <div className={`${container} overflow-hidden w-100`}>
          <div
            className={`${fill} shadow-2`}
            style={{ width: `${clamp(percent, 0, 100)}%` }}
          />
        </div>
      </div>
    </div>
  );
};

ProgressBar.propTypes = {
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  large: PropTypes.bool,
  percent: PropTypes.number,
  pill: PropTypes.bool,
};

ProgressBar.defaultProps = {
  backgroundColor: 'near-white',
  color: 'black',
  large: false,
  percent: 0,
  pill: false,
};

export default ProgressBar;
