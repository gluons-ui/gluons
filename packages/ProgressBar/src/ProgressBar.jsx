import React from 'react';
import PropTypes from 'prop-types';
import clamp from 'lodash/clamp';

const ProgressBar = ({
  background,
  color,
  large,
  percentage,
  rounded }) => {
  const isLarge = `${large ? 'h2' : 'h1'}`;
  const isRounded = `${rounded ? 'br-pill' : ''}`;
  const fill = `bg-${color} ${isLarge} ${isRounded}`;
  const container = `bg-${background} ${isLarge} ${isRounded}`;
  return (
    <div className="progress-bar w-100">
      <div className="w-100">
        <div className={`${container} overflow-hidden w-100`}>
          <div
            className={`${fill} shadow-2`}
            style={{ width: `${clamp(percentage, 0, 100)}%` }}
          />
        </div>
      </div>
    </div>
  );
};

ProgressBar.propTypes = {
  background: PropTypes.string,
  color: PropTypes.string,
  large: PropTypes.bool,
  percentage: PropTypes.number,
  rounded: PropTypes.bool,
};

ProgressBar.defaultProps = {
  background: 'light-gray',
  color: 'gray',
  large: false,
  percentage: 50,
  rounded: false,
};

export default ProgressBar;
