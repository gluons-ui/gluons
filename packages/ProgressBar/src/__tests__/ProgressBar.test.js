import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProgressBar from '../ProgressBar';

configure({ adapter: new Adapter() });

describe('<ProgressBar />', () => {
  it('renders defaults', () => {
    expect(shallow(<ProgressBar />)).toMatchSnapshot();
  });

  it('renders custom color', () => {
    expect(shallow(<ProgressBar color="blue" />)).toMatchSnapshot();
  });

  it('renders custom background color', () => {
    expect(shallow(<ProgressBar color="blue" />)).toMatchSnapshot();
  });

  it('renders large version correctly', () => {
    expect(shallow(<ProgressBar large />)).toMatchSnapshot();
  });

  it('renders rounded version correctly', () => {
    expect(shallow(<ProgressBar rounded />)).toMatchSnapshot();
  });

  it('renders 0% correctly', () => {
    expect(shallow(<ProgressBar percentage={0} />)).toMatchSnapshot();
  });

  it('renders 100% correctly', () => {
    expect(shallow(<ProgressBar percentage={100} />)).toMatchSnapshot();
  });
});
