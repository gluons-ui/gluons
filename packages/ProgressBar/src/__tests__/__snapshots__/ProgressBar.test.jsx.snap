// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders 0% correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    background="light-gray"
    color="gray"
    large={false}
    percentage={0}
    rounded={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w-100"
      >
        <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>
      </div>,
      "className": "progress-bar w-100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "0%",
              }
            }
          />
        </div>,
        "className": "w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "0%",
              }
            }
          />,
          "className": "bg-light-gray h1  overflow-hidden w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-gray h1  shadow-2",
            "style": Object {
              "width": "0%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w-100"
        >
          <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "0%",
                }
              }
            />
          </div>
        </div>,
        "className": "progress-bar w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "0%",
                }
              }
            />
          </div>,
          "className": "w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "0%",
                }
              }
            />,
            "className": "bg-light-gray h1  overflow-hidden w-100",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-gray h1  shadow-2",
              "style": Object {
                "width": "0%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ProgressBar /> renders 100% correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    background="light-gray"
    color="gray"
    large={false}
    percentage={100}
    rounded={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w-100"
      >
        <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </div>,
      "className": "progress-bar w-100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>,
        "className": "w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "100%",
              }
            }
          />,
          "className": "bg-light-gray h1  overflow-hidden w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-gray h1  shadow-2",
            "style": Object {
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w-100"
        >
          <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </div>
        </div>,
        "className": "progress-bar w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "100%",
                }
              }
            />
          </div>,
          "className": "w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "100%",
                }
              }
            />,
            "className": "bg-light-gray h1  overflow-hidden w-100",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-gray h1  shadow-2",
              "style": Object {
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ProgressBar /> renders custom background color 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    background="light-gray"
    color="blue"
    large={false}
    percentage={50}
    rounded={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w-100"
      >
        <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-blue h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>,
      "className": "progress-bar w-100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-blue h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>,
        "className": "w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-blue h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />,
          "className": "bg-light-gray h1  overflow-hidden w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-blue h1  shadow-2",
            "style": Object {
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w-100"
        >
          <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-blue h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>
        </div>,
        "className": "progress-bar w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-blue h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>,
          "className": "w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="bg-blue h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />,
            "className": "bg-light-gray h1  overflow-hidden w-100",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-blue h1  shadow-2",
              "style": Object {
                "width": "50%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ProgressBar /> renders custom color 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    background="light-gray"
    color="blue"
    large={false}
    percentage={50}
    rounded={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w-100"
      >
        <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-blue h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>,
      "className": "progress-bar w-100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-blue h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>,
        "className": "w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-blue h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />,
          "className": "bg-light-gray h1  overflow-hidden w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-blue h1  shadow-2",
            "style": Object {
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w-100"
        >
          <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-blue h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>
        </div>,
        "className": "progress-bar w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-blue h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>,
          "className": "w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="bg-blue h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />,
            "className": "bg-light-gray h1  overflow-hidden w-100",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-blue h1  shadow-2",
              "style": Object {
                "width": "50%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ProgressBar /> renders defaults 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    background="light-gray"
    color="gray"
    large={false}
    percentage={50}
    rounded={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w-100"
      >
        <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>,
      "className": "progress-bar w-100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bg-light-gray h1  overflow-hidden w-100"
        >
          <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>,
        "className": "w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-gray h1  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />,
          "className": "bg-light-gray h1  overflow-hidden w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-gray h1  shadow-2",
            "style": Object {
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w-100"
        >
          <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>
        </div>,
        "className": "progress-bar w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-light-gray h1  overflow-hidden w-100"
          >
            <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>,
          "className": "w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="bg-gray h1  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />,
            "className": "bg-light-gray h1  overflow-hidden w-100",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-gray h1  shadow-2",
              "style": Object {
                "width": "50%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ProgressBar /> renders large version correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    background="light-gray"
    color="gray"
    large={true}
    percentage={50}
    rounded={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w-100"
      >
        <div
          className="bg-light-gray h2  overflow-hidden w-100"
        >
          <div
            className="bg-gray h2  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>,
      "className": "progress-bar w-100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bg-light-gray h2  overflow-hidden w-100"
        >
          <div
            className="bg-gray h2  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>,
        "className": "w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-gray h2  shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />,
          "className": "bg-light-gray h2  overflow-hidden w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-gray h2  shadow-2",
            "style": Object {
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w-100"
        >
          <div
            className="bg-light-gray h2  overflow-hidden w-100"
          >
            <div
              className="bg-gray h2  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>
        </div>,
        "className": "progress-bar w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-light-gray h2  overflow-hidden w-100"
          >
            <div
              className="bg-gray h2  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>,
          "className": "w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="bg-gray h2  shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />,
            "className": "bg-light-gray h2  overflow-hidden w-100",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-gray h2  shadow-2",
              "style": Object {
                "width": "50%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`<ProgressBar /> renders rounded version correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBar
    background="light-gray"
    color="gray"
    large={false}
    percentage={50}
    rounded={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="w-100"
      >
        <div
          className="bg-light-gray h1 br-pill overflow-hidden w-100"
        >
          <div
            className="bg-gray h1 br-pill shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>
      </div>,
      "className": "progress-bar w-100",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="bg-light-gray h1 br-pill overflow-hidden w-100"
        >
          <div
            className="bg-gray h1 br-pill shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />
        </div>,
        "className": "w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-gray h1 br-pill shadow-2"
            style={
              Object {
                "width": "50%",
              }
            }
          />,
          "className": "bg-light-gray h1 br-pill overflow-hidden w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-gray h1 br-pill shadow-2",
            "style": Object {
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w-100"
        >
          <div
            className="bg-light-gray h1 br-pill overflow-hidden w-100"
          >
            <div
              className="bg-gray h1 br-pill shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>
        </div>,
        "className": "progress-bar w-100",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="bg-light-gray h1 br-pill overflow-hidden w-100"
          >
            <div
              className="bg-gray h1 br-pill shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />
          </div>,
          "className": "w-100",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="bg-gray h1 br-pill shadow-2"
              style={
                Object {
                  "width": "50%",
                }
              }
            />,
            "className": "bg-light-gray h1 br-pill overflow-hidden w-100",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bg-gray h1 br-pill shadow-2",
              "style": Object {
                "width": "50%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
